custom_components/unifiprotect/binary_sensor.py

```diff
@@ -180,64 +240,115 @@ def _async_fire_events(self):
+            ):
+                obj = [
+                    t.value
+                    for t in self.device.last_smart_detect_event.smart_detect_types
+                ]
+        else:
+            is_on = self.device.is_pir_motion_detected
+
+        self.hass.bus.async_fire(
+            f"{DOMAIN}_motion",
+            {
+                "entity_id": self.entity_id,
+                "smart_detect": obj,
+                "motion_on": is_on,
+            },
+        )
+
     @property
     def extra_state_attributes(self):
         """Return the extra state attributes."""
         key = self.entity_description.key
         attr = {
             **super().extra_state_attributes,
         }
-        if key == DEVICE_TYPE_DARK:
+        if key == _KEY_DARK:
             return attr
-        if key == DEVICE_TYPE_DOORBELL:
-            attr[ATTR_LAST_TRIP_TIME] = self._device_data["last_ring"]
+        if key == _KEY_DOORBELL:
+            attr[ATTR_LAST_TRIP_TIME] = self.device.last_ring.isoformat()
             return attr
 
-        if self._device_data["type"] == DEVICE_TYPE_SENSOR:
-            if key == DEVICE_TYPE_MOTION:
-                attr[ATTR_LAST_TRIP_TIME] = self._device_data["last_motion"]
-                attr[ATTR_EVENT_LENGTH] = self._device_data["event_length"]
-            if key == "door":
-                attr[ATTR_LAST_TRIP_TIME] = self._device_data["last_openchange"]
-                attr[ATTR_EVENT_LENGTH] = self._device_data["event_length"]
+        if isinstance(self.device, Sensor):
+            if key in (_KEY_MOTION, _KEY_DOOR):
+                if key == _KEY_MOTION:
+                    last_trip = self.device.motion_detected_at
+                else:
+                    last_trip = self.device.open_status_changed_at
+
+                attr[ATTR_LAST_TRIP_TIME] = last_trip.isoformat()
+            return attr
+        elif isinstance(self.device, Light):
+            if key == _KEY_MOTION:
+                attr[ATTR_LAST_TRIP_TIME] = self.device.last_motion.isoformat()
             return attr
 
         # Camera motion sensors with object detection
+        score = 0
         if (
-            self._device_data["event_object"] is not None
-            and len(self._device_data["event_object"]) > 0
+            self.device.is_smart_detected
+            and self.device.last_smart_detect_event is not None
+            and len(self.device.last_smart_detect_event.smart_detect_types) > 0
         ):
-            detected_object = self._device_data["event_object"][0]
+            score = self.device.last_smart_detect_event.score
+            detected_object = self.device.last_smart_detect_event.smart_detect_types[0]
             _LOGGER.debug(
-                "OBJECTS: %s on %s", self._device_data["event_object"], self._attr_name
+                "OBJECTS: %s on %s",
+                self.device.last_smart_detect_event.smart_detect_types,
+                self._attr_name,
             )
```

Comment:
This is existing, but side effects aren't allowed in `property` methods

Reply:
I can refactor that to be cleaner.

Reply:
I moved all of the code in this property to a separate method that is called by `_async_update_device_from_protect`. Then this just uses those values (merging them with the parent ones).